<launch>
  <!-- Launch file for velodyne16 VLP-16 LiDAR -->

    <arg name="rviz" default="true" />

    <arg name="map" value="$(find fast_lio)/PCD/filtered_scans.pcd"/>

    <node name="map_child_tf_publisher" pkg="tf" type="static_transform_publisher"
          args="0 0 0 0 0 0 map map_child 100"/>
    
    <!-- <node name="base_body_tf_publisher" pkg="tf" type="static_transform_publisher"
          args="0 0 0 0 0 0 body velodyne 100"/> -->

    <node name="map_publisher" pkg="fast_lio" type="map_publisher" output="screen">
        <param name="map_path" value="$(arg map)"/>
    </node>

    <node name="fast_lio_localization" pkg="fast_lio" type="fastlio_localization" output="screen">
        <param name="ndt/debug" value="true"/>
        <param name="ndt/voxel_leaf_size" value="0.1"/>
        <param name="ndt/min_scan_range" value="1.0"/>
    </node>

    <rosparam command="load" file="$(find fast_lio)/config/velodyne_localization.yaml" />

    <param name="feature_extract_enable" type="bool" value="0"/>
    <param name="point_filter_num" type="int" value="4"/>
    <param name="max_iteration" type="int" value="3" />
    <param name="filter_size_surf" type="double" value="0.5" />
    <param name="filter_size_map" type="double" value="0.5" />
    <param name="cube_side_length" type="double" value="1000" />
    <param name="runtime_pos_log_enable" type="bool" value="0" />
    <node pkg="fast_lio" type="fastlio_odometry" name="laserOdometry" output="screen" /> 

    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio)/rviz_cfg/localization.rviz" />
    </group>

</launch>